### PR TITLE
Switch from supervisord to systemd

### LINKED ISSUE #18: [IMPROVEMENT] Upgrade Amazon Linux Support to Version 2
AWS update there Amazon Linux to a new version 2: https://aws.amazon.com/de/amazon-linux-2/

**Interesting Features:**
- systemd
- on-premise support ( Docker, ... )

**Outlook:**
The support of systemd would facilitate the setup and configuration of Nomad and Consul, as well.
Supervisor would not be needed anymore to streamline setup, since Debian and Ubuntu are using systemd as well.

**Question:**
Should the Amazon Linux 1 support be dropped?
1. Pro: Legacy support
2. Contra: Duplication of many configuration files, especially for the AMI creation.
3. Contra: Increased complexity in the `install-nomad` and `run-nomad` script.

_Suggestion:_
Drop support since the provided setup can only be seen as an example.

_Note:_
Once this is decided, I can provide a PR.

### PR DESCRIPTION
**Pull request by draoncc**
Closes #18 by switching to Amazon Linux 2 and changing to `systemd`.

Heavily based on https://github.com/hashicorp/terraform-aws-consul/pull/130/files

After noticing the lack of activity on #45 I decided to create a pull request myself. All points brought up in that stale pull request should have been addressed. I have so far been unable to get the tests running, though.

### CODE REVIEW

File: modules/run-nomad/README.md [hunk 1]
```diff
@@ -34,18 +34,24 @@ This will:
...
 when the EC2 Instance is first booting. If you are running Consul on the same server, make sure to use this script
-*after* Consul has booted. After running `run-nomad` on that initial boot, the `supervisord` configuration
+*after* Consul has booted. After running `run-nomad` on that initial boot, the `systemd` configuration
 will automatically restart Nomad if it crashes or the EC2 instance reboots.
 
+Note that `systemd` logs to its own journal by default.  To view the Nomad logs, run `journalctl -u nomad.service`.  To change
+the log output location, you can specify the `StandardOutput` and `StandardError` options by using the `--systemd-stdout` and `--systemd-stderr`
+options.  See the [`systemd.exec` man pages](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#StandardOutput=) for available
+options, but note that the `file:path` option requires [systemd version >= 236](https://stackoverflow.com/a/48052152), which is not provided 
+in the base Ubuntu 16.04 and Amazon Linux 2 images.
```
Comment (brikis98) [inline modules/run-nomad/README.md:53]: Thanks for updating the docs 👍

**Comment by brikis98** [comment]
Packer build failed:

```
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami: Or undo this operation with:
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami:
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami:   git switch -
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami:
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami: Turn off this advice by setting config variable advice.detachedHead to false
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami:
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami: 2020-04-08 08:34:04 [INFO] [install-consul] Starting Consul install
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:04Z command.go:158: 1586334844,,ui,error,==> amazon-linux-2-ami: 2020-04-08 08:34:04 [INFO] [install-consul] Installing dependencies
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:05Z command.go:158: 1586334845,,ui,message,    amazon-linux-2-ami: Loaded plugins: extras_suggestions%!(PACKER_COMMA) langpacks%!(PACKER_COMMA) priorities%!(PACKER_COMMA) update-motd
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:05Z command.go:158: 1586334845,,ui,message,    amazon-linux-2-ami: No packages marked for update
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:05Z command.go:158: 1586334845,,ui,message,    amazon-linux-2-ami: Loaded plugins: extras_suggestions%!(PACKER_COMMA) langpacks%!(PACKER_COMMA) priorities%!(PACKER_COMMA) update-motd
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:05Z command.go:158: 1586334845,,ui,message,    amazon-linux-2-ami: No package aws available.
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:06Z command.go:158: 1586334846,,ui,message,    amazon-linux-2-ami: Package curl-7.61.1-12.amzn2.0.1.x86_64 already installed and latest version
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:06Z command.go:158: 1586334846,,ui,message,    amazon-linux-2-ami: Package unzip-6.0-20.amzn2.x86_64 already installed and latest version
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:06Z command.go:158: 1586334846,,ui,message,    amazon-linux-2-ami: Package jq-1.5-1.amzn2.0.2.x86_64 already installed and latest version
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:06Z command.go:158: 1586334846,,ui,message,    amazon-linux-2-ami: Nothing to do
TestNomadConsulClusterSeparateAmazonLinux2Ami 2020-04-08T08:34:06Z command.go:158: 1586334846,,ui,error,==> amazon-linux-2-ami: sudo: pip: command not found
```

**Comment by draoncc** [comment]
It seems I forgot to update `consul_module_version` in the packer build files. That's fixed now, though the earliest version of that module that builds for Amazon Linux 2 also installs consul with version 1.2.2, whereas this module still goes with 1.0.3.

I've updated the Consul version to 1.5.1, to match the latest version of [terraform-aws-consul](https://github.com/hashicorp/terraform-aws-consul/blob/v0.7.4/examples/consul-ami/consul.json), and Nomad to the nearest version going by release date, which is 0.9.1.

**Comment by brikis98** [comment]
Great, thanks! I'll kick off tests again.

**Comment by brikis98** [comment]
Alright, tests passed! Merging now. Thank you for the contribution!

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-nomad/releases/tag/v0.6.0

**Pull request merged**